supla-common/proto.h

```diff
@@ -2064,24 +2246,128 @@ typedef struct {
   _supla_int_t Param3;
 } TSD_ChannelIntParams;
 
-#define SUPLA_CHANNEL_CONFIG_MAXSIZE 128
+#define SUPLA_CHANNEL_CONFIG_MAXSIZE 512  // v. <= 19 - 128; v. >= 20 - 512
+#define SUPLA_DEVICE_CONFIG_MAXSIZE 512
+
+// Default type is used for standard channel config (user modifiable)
 #define SUPLA_CONFIG_TYPE_DEFAULT 0
+// Weekly schedule
+#define SUPLA_CONFIG_TYPE_WEEKLY_SCHEDULE 2
+
+/********************************************
+ * DEVICE CONFIG STRUCTURES
+ ********************************************/
+
+// SUPLA_DS_CALL_SET_DEVICE_CONFIG
+// SUPLA_SD_CALL_SET_DEVICE_CONFIG
+// Config field should contain fields stored in order as they appear in
+// Fields parameter. Size of parameter depends on Field type.
+typedef struct {
+  unsigned char EndOfDataFlag;  // 1 - last message; 0 - more messages will come
+  unsigned char zero[8];  // for future use
+  unsigned _supla_int64_t Fields;   // bit map of SUPLA_DEVICE_CONFIG_FIELD_
+  unsigned _supla_int16_t ConfigSize;
+  char Config[SUPLA_DEVICE_CONFIG_MAXSIZE];  // Last variable in struct!
+} TSD_SetDeviceConfig;  // v. >= 20
 
+// SUPLA_SD_CALL_SET_DEVICE_CONFIG_RESULT
+// SUPLA_DS_CALL_SET_DEVICE_CONFIG_RESULT
+typedef struct {
+  unsigned char Result;  // SUPLA_CONFIG_RESULT_*
+  unsigned char ConfigType;  // SUPLA_DEVICE_CONFIG_TYPE_*
```

Comment:
Shouldn't ConfigType be in TSD_SetDeviceConfig too?

Reply:
Probably I forgot to remove it, since you requested to do so earlier.